internal/model/todos.go

```diff
@@ -10,43 +14,42 @@ type Todo struct {
 
 // TodoList kind of a has many struct
 type TodoList struct {
-	ID    string `json:"id"`
-	Name  string `json:"name"`
-	Todos []Todo `json:"todos"`
+	ID    string  `json:"id"`
+	Name  string  `json:"name"`
+	Todos []*Todo `json:"todos"`
 }
 
 // GetTodosListByID returns slice of Todos based on the listID passed in
-func GetTodosListByID(listID string) (TodoList, error) {
+func GetTodosListByID(listID string) (*TodoList, error) {
 	listStatement := `SELECT id, name FROM lists WHERE id = $1`
-	list := TodoList{}
+	list := &TodoList{}
 
 	err := db.QueryRow(listStatement, listID).Scan(&list.ID, &list.Name)
 	if err != nil {
-		return list, err
+		return nil, err
```

Comment:
What's also nice, is now that we're using a pointer return value, we can use nil as a return. That way if someone ignores our error we're sure they'll run into issues because instead of an empty TodoList they'll get an error that the list was nil.

Makes writing these returns a lot cleaner too, right?